Source/WebCore/crypto/gcrypt/CryptoKeyOKPGCrypt.cpp

```diff
@@ -63,52 +63,77 @@ static bool supportedAlgorithmIdentifier(CryptoAlgorithmIdentifier keyIdentifier
     return false;
 }
 
+Vector<uint8_t> nine {
+    0x09, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+};
+
 }
 
-std::optional<CryptoKeyPair> CryptoKeyOKP::platformGeneratePair(CryptoAlgorithmIdentifier identifier, NamedCurve namedCurve, bool extractable, CryptoKeyUsageBitmap usages)
+static void gcryptGenerateEd25519Keys(Vector<uint8_t>&& publicKeyData, Vector<uint8_t>&& privateKeyData)
```

Comment:
This isn't correct usage of rvalue references, but I would recommend a different interface for these two functions:

```
static std::optional<std::pair<Vector<uint8_t>, Vector<uint8_t>>> gcryptGenerateEd25519Keys();
```

This way you return std::nullopt on any error that might occur during generation, and otherwise you return the pair of keys.

Reply:
Done